File: csharp/TellDontAsk.Tests/Doubles/TestOrderRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TellDontAsk.Domain;
using TellDontAsk.Repository;

namespace TellDontAsk.Tests.Doubles
{
    public class TestOrderRepository : OrderRepository
    {
        private Order insertedOrder;
        private List<Order> orders = new List<Order>();

        public Order getSavedOrder()
        {
            return insertedOrder;
        }

        public void save(Order order)
        {
            this.insertedOrder = order;
        }

        public Order getById(int orderId)
        {
            return orders.FirstOrDefault(o => o.getId() == orderId);
        }

        public void addOrder(Order order)
        {
            this.orders.Add(order);
        }
    }
}

File: php/tests/Doubles/TestOrderRepository.php
<?php

namespace Archel\TellDontAskTest\Doubles;

use Archel\TellDontAsk\Domain\Order;
use Archel\TellDontAsk\Repository\OrderRepository;

/**
 * Class TestOrderRepository
 * @package Archel\TellDontAskTest\Doubles
 */
class TestOrderRepository implements OrderRepository
{
    /**
     * @var Order
     */
    private $insertedOrder;

    /**
     * @var array
     */
    private $orders = [];

    /**
     * @param Order $order
     */
    public function save(Order $order): void
    {
        $this->insertedOrder = $order;
    }

    /**
     * @param int $orderId
     * @return Order
     */
    public function getById(int $orderId): ?Order
    {
        $order = array_filter($this->orders, function ($order) use ($orderId) {
            return $order->getId() === $orderId;
        });

        return !empty($order) ? current($order) : null;
    }

    /**
     * @param Order $order
     */
    public function addOrder(Order $order) : void
    {
        $this->orders[] = $order;
    }

    /**
     * @return Order
     */
    public function getSavedOrder() : Order
    {
        return $this->insertedOrder;
    }
}